counsel.el

```diff
@@ -3836,17 +3836,31 @@ include attachments of other Org buffers."
   (require 'org-capture)
   (ivy-read "Capture template: "
             (delq nil
-                  (mapcar
-                   (lambda (x)
-                     (when (> (length x) 2)
-                       (format "%-5s %s" (nth 0 x) (nth 1 x))))
-                   ;; We build the list of capture templates as in
-                   ;; `org-capture-select-template':
-                   (or (org-contextualize-keys
-                        (org-capture-upgrade-templates org-capture-templates)
-                        org-capture-templates-contexts)
-                       '(("t" "Task" entry (file+headline "" "Tasks")
-                          "* TODO %?\n  %u\n  %a")))))
+                  (let (messages prefixes)
+                    ;; We build the list of capture templates as in
+                    ;; `org-capture-select-template':
+                    (dolist (x (or (org-contextualize-keys
+                                    (org-capture-upgrade-templates org-capture-templates)
+                                    org-capture-templates-contexts)
+                                   '(("t" "Task" entry (file+headline "" "Tasks")
+                                      "* TODO %?\n  %u\n  %a")))
+                               (reverse messages))
```

Comment:
Since you are building `messages` yourself, you can use `nreverse` instead of `reverse`.

Also, please don't use `dolist`'s third `RESULT` argument as it doesn't play nicely with byte-compilation and is considered passé.

Why not rewrite the current `(delq nil (dolist ...))` structure more cleanly using `cl-mapcan` instead?

Reply:
Thanks for the suggestions. I cleaned it up and used `cl-reduce` instead of `cl-mapcan` because I needed a way to carry the prefixes found over to the next loop.